proposals/nnnn-sdl-policy-server-enhancements.md

```diff
@@ -0,0 +1,56 @@
+
+## Motivation
+The current policy server example on Github is severely limited in functionality. The current implementation raises many questions by OEMs looking to host their own copy of a policy server. The policy server example currently lacks integration with SHAID, and fails to offer an intuitive way to change any aspect of the policy table. The proposed changes below will address these concerns.
+
+## Proposed solution
+
+#### About
+The policy server will be written in NodeJS. It will replace the old sdl_server project currently on Github. The release of this new policy server will be developed over multiple stages. This proposal describes only the initial stage of the project; more proposals will follow that build out the project (front end UI, encryption, etc).
+
+#### SHAID Integration
+The server will communicate with SHAID through its APIs.  This includes returning the initial list of apps during setup. It also includes getting alerts when applications have been modified as reported by through the developer portal. The authentication keys, accessible via the developer portal, can be used to authenticate the policy server with SHAID. A wrapper for calling the SHAID API will be available as a node module on the npm registry. Anyone will be able to download it for use, and the SDL Policy Server will use it to interact with SHAID. 
+
+#### Data Storage and Versioning
+The server will be able to connect to a SQL database by default for storing application version history, policy table component history, and other data. It will be designed to easily support multiple SQL database engines by being limited to using basic SQL commands. 
+
+There will be database migration scripts which allow easy use of setting up and tearing down the tables. There will also be initial data created that is necessary for the policy server to function. Static tables such as languages, RPC names and vehicle data component names will be defined, and initial data creation will include consumer friendly messages and the module config from the default policy table. Another piece of the initial data will be generated through SHAID API calls, which include category names and country information. The pieces of the policy table stored will be designed to be extendable for other OEMs who wish to add their own information to their database. The structure of the tables will be documented and posted on the policy server github.
+
+Lastly, the versioning system will have two independent fields. There will be a unique ID for data such as the module config object and the function groups object so that many different objects can be created. Secondly, there will be a status field for these versioned objects that determine whether they should be exist on production or in a testing state. Therefore, there will be a staging version and a production version of a policy table that can be used.
```

Comment:
Please update third sentence from "...whether they should be exist on production..." to "whether they should exist on production..."